src/funcs.c

```diff
@@ -2825,3 +2825,71 @@ void joy_right()
   set(app->LV_GamesList, MUIA_List_Active, ind-1);
 
 }
+
+ULONG get_wb_version()
+{
+	static ULONG ver = 0UL;
+
+	if (ver != 0UL)
+	{
+		return ver;
+	}
+
+	if (WorkbenchBase == NULL)
+	{
+		//Really Workbench library should be already opened
+		// Somehow we're running without Workbench.
+		// Nothing to do since the version variable inits with 0.
+		return ver;
+	}
+
+	// Save workbench.library version for later calls
+	ver = WorkbenchBase->lib_Version;
+
+	return ver;
+}
+
+void open_current_dir()
+{
+	if (get_wb_version() < 44)
+	{
+		// workbench.library doesn't support OpenWorkbenchObjectA yet
+		return;
+	}
+
+	// Allocate Memory for variables
+	char* game_title = NULL;
+
+	//set the elements on the window
+	DoMethod(app->LV_GamesList, MUIM_List_GetEntry, MUIV_List_GetEntry_Active, &game_title);
+	if (game_title == NULL || strlen(game_title) == 0)
+	{
+		msg_box((const char*)GetMBString(MSG_SelectGameFromList));
+		return;
+	}
+
+	const int found = title_exists(game_title);
+	if (!found)
+	{
+		msg_box((const char*)GetMBString(MSG_SelectGameFromList));
+		return;
+	}
+	
+	int full_path_length;
+	int name_length;
+	int path_length;
+	full_path_length = strlen(item_games->path);
+	name_length = strlen(PathPart(item_games->path));
+	path_length = (full_path_length - name_length)+1;
+	
+	char *path_only = AllocVec(path_length, MEMF_ANY);		//reserve memory for path
+	if(!path_only) Printf("no memory?\n");
```

Comment:
Where will the user see this message? Should we inform the user in a better way?

Reply:
I'm pretty certain that a console window will appear with the message in it, but that's only if there's memory available for it to begin with. Points to Tolkien for considering the error case though, considering that elsewhere through funcs.c there are numerous uses of `malloc()` without any kind of check that it's not returning NULL.

Reply:
I see... I am not sure though that a console will appear, because on some of my tests never happened. I had to start the app from shell to see those printfs.

Reply:
Sorry. that was a test of mine. I'm learning to code and needed to know if it was right.
I think we can remove it or perhaps replace it forma a msgbox.

Reply:
In an environment where there isn't enough free contiguous memory to allocate space for a string to contain the drawer path, I doubt there'd be enough memory to create a console or a message box either. As I'll be updating the code to use `get_directory_name()` after work today, this'll be moot anyway.